enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,355 @@
+---
+title: ingress-node-firewall-with-ebpf
+authors:
+  - "@msherif1234"
+reviewers:
+  - "@danwinship"
+  - "@dave-tucker"
+  - "@dcbw"
+  - "@trozet"
+approvers:
+  - "@danwinship"
+  - "@knobunc"
+api-approvers:
+  - "@danwinship"
+creation-date: 2022-06-15
+last-updated: 2022-06-15
+status: implementable
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-2781
+---
+
+# Ingress node Firewall using eBPF
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OCP customers would like to secure OCP's nodes from external DDoS attacks by configuring ingress firewall policies.
+eBPF provides very flexible filtering mechansim to allow early detection and filtering for a wide range of protocols IPv4/IPv6/ICMP/TCP/UDP/ICMPv6/SCTP.
+
+New cluster scoped CRD `GolbalNodeFirewall` will be used to configure node ingress firewall match rules and a per matching rule statistics will be avaliable.
+
+## Motivation
+
+Customers would like to protect OCP cluster from external DDoS attacks.
+
+### Goals
+
+- Solution based on using `eBPF` for packet filtering.
+- flexiable match filters rules for customer to use with a CRD validation to detect and fail invalid configuration.
+- Provide a mechanism for customers to select which nodes via `nodeSelector` and the node's interface(s) where the firewall policy will be attached to.
+- Ingress node firewall is deployed on a fully operational OCP cluster (i.e. cluster comes up with no default ingress firewall policy).
+- fail safe in case of user provisioned invalid rules which can lockup cluster operation.
+
+### Non-Goals
+
+## Proposal
+
+- customers will label nodes where ingress node policy will be deployed on.
+- customers will prepare `GlobalNodeFirewall` object with all required firewall rules and deploy.
+- IngressNodeFW controller will process `GlobalNodeFirewall` events and generate eBPF configuration rules and interfaces in configuration files at a specific location in the kubernetes node filesystem.
+- NodeFw's pod will have an initContainer that will load and attach eBPF kernel hook loaded as `BPF_PROG_TYPE_XDP` to the list of interfaces configured in `GlobalNodeFirewall` object.
+- NodeFw's pod manager container will be watching those configuration files and use it to update eBPF ingress table Maps which is shared between userspace and kernel.
```

Comment:
true the CNO controller that manage the new CRD can also be the eBPF loader the only reason I originally pulled it out in case we need to explore with bpfd or cilium ebpf didn't want to overwhelm CNO 

so the controller now need to have :
- verification webhook to verify config and do the fail safe checks
- handle CRD add/update/delete
- load/update/delete eBPF table

Reply:
I think having separate olm operator give us much more flexibility vs. running in CNO we can bundle api, verification webhook, reconciler, eBPF loader and e2e all together in new d/s repo